src/animals/animals.service.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  BadRequestException,
+  HttpException,
+  HttpStatus,
+  Injectable,
+} from '@nestjs/common';
+import { AnimalDto, CreateAnimalDto, UpdateAnimalDto } from '@animals/dto';
+import { v4 as uuidv4 } from 'uuid';
+import { animals } from '@animals/animals';
+import { PaginationDto } from 'pagination/dto/pagination.dto';
+
+const maxAnimalsPerUser = 10;
+
+@Injectable()
+export class AnimalsService {
+  getAll(pagination: PaginationDto, userId: string) {
+    return animals
+      .filter((animal) => animal.ownerId == userId)
+      .slice(pagination.offset, pagination.offset + pagination.limit);
+  }
+
+  createAnimal(createAnimalDto: CreateAnimalDto, userId: string) {
+    const currentNumberOfUserAnimals = animals.filter(
+      (animal) => animal.ownerId === userId,
+    ).length;
+    if (currentNumberOfUserAnimals >= maxAnimalsPerUser) {
+      throw new HttpException(
```

Comment:
You can throw new BadRequestException, that inplemented in nest

Reply:
Fixed